Hippo.Web/ClientApp/src/components/Order/Details.tsx

```diff
@@ -0,0 +1,685 @@
+import React, { useCallback, useContext, useEffect, useState } from "react";
+import { Link, useParams } from "react-router-dom";
+import {
+  HistoryModel,
+  OrderModel,
+  PaymentModel,
+  UpdateOrderStatusModel,
+} from "../../types";
+import { authenticatedFetch, parseBadRequest } from "../../util/api";
+import { ReactTable } from "../../Shared/ReactTable";
+import { createColumnHelper } from "@tanstack/react-table";
+import OrderForm from "./OrderForm";
+import { usePermissions } from "../../Shared/usePermissions";
+import { useConfirmationDialog } from "../../Shared/ConfirmationDialog";
+import { usePromiseNotification } from "../../util/Notifications";
+import { notEmptyOrFalsey } from "../../util/ValueChecks";
+import { ShowFor } from "../../Shared/ShowFor";
+import AppContext from "../../Shared/AppContext";
+import {
+  convertToPacificDate,
+  convertToPacificTime,
+} from "../../util/DateHelper";
+
+export const Details = () => {
+  const { cluster, orderId } = useParams();
+  const [{ user }] = useContext(AppContext);
+  const [order, setOrder] = useState<OrderModel | null>(null);
+  const [balanceRemaining, setBalanceRemaining] = useState<number>(0);
+  const [balancePending, setBalancePending] = useState<number>(0);
+  const { isClusterAdminForCluster } = usePermissions();
+  const [isClusterAdmin, setIsClusterAdmin] = useState(null);
+  const [notification, setNotification] = usePromiseNotification();
+  const [updateStatusModel, setUpdateStatusModel] =
+    useState<UpdateOrderStatusModel | null>(null);
+  const adminEditableStatuses = ["Processing", "Active"];
+  const sponsorEditableStatuses = ["Created"];
+
+  useEffect(() => {
+    setIsClusterAdmin(isClusterAdminForCluster());
+  }, [isClusterAdmin, isClusterAdminForCluster]);
+
+  const calculateBalanceRemaining = (data: any) => {
```

Comment:
do we need any here?  would be nice to have a real type

Reply:
Maybe? the parseFloat below seems to have problems if I do.

Reply:
#391